packages/circuit-ui/components/Popover/Popover.spec.tsx

```diff
@@ -0,0 +1,224 @@
+    });
+  });
+
+  describe('accessibility', () => {
+    it('should meet accessibility guidelines', async () => {
+      const props = {
+        ...baseProps,
+        href: 'https://sumup.com',
+        onClick: jest.fn(),
+        icon: Zap,
+      };
+      const wrapper = renderPopoverItem(renderToHtml, props);
+      const actual = await axe(wrapper);
+      expect(actual).toHaveNoViolations();
+    });
+  });
+});
+
+describe('Popover', () => {
+  const Default = (props) => {
+    const referenceElement = useRef<HTMLButtonElement & HTMLAnchorElement>(
+      null,
+    );
+
+    return (
+      <>
+        <button ref={referenceElement}>Button</button>
+        <Popover referenceElement={referenceElement} {...props} />
+      </>
+    );
+  };
+
+  const Interactive = () => {
+    const [isOpen, setOpen] = useState(true);
+    const referenceElement = useRef<HTMLButtonElement & HTMLAnchorElement>(
+      null,
+    );
+
+    const handleClick = () => {
+      setOpen((prev) => !prev);
+    };
+
+    const onClose = () => {
+      setOpen(false);
+    };
+
+    return (
+      <>
+        <Button
+          size="kilo"
+          variant="tertiary"
+          onClick={handleClick}
+          ref={referenceElement}
+          icon={ThumbUp}
+        >
+          Button
+        </Button>
+        <Popover
+          actions={[
+            {
+              onClick: () => alert('Hello'),
+              children: 'Add',
+              icon: CirclePlus,
+            },
+            {
+              onClick: () => alert('Hello'),
+              children: 'Edit',
+              icon: PenStroke,
+            },
+            { type: 'divider' },
+            {
+              onClick: () => alert('Hello'),
+              children: 'Delete',
+              icon: Bin,
+              destructive: true,
+            },
+          ]}
+          onClose={onClose}
+          isOpen={isOpen}
+          referenceElement={referenceElement}
+        />
+      </>
+    );
+  };
+
+  describe('styles', () => {
+    it('should render with default styles', () => {
+      const props = {
+        actions: [
+          {
+            onClick: () => alert('Hello'),
+            children: 'Add',
+            icon: CirclePlus,
+          },
+        ],
+        isOpen: true,
+        onClose: jest.fn(),
+      };
+
+      const actual = create(<Default {...props} />);
+      expect(actual).toMatchSnapshot();
+    });
+
+    placements.forEach((placement) => {
+      it(`should render popover on ${placement}`, () => {
+        const props = {
+          actions: [
+            {
+              onClick: () => alert('Hello'),
+              children: 'Add',
+              icon: CirclePlus,
+            },
+          ],
+          isOpen: true,
+          onClose: jest.fn(),
+        };
+        const actual = create(<Default {...props} />);
+        expect(actual).toMatchSnapshot();
+      });
+    });
+
+    it('should render popover when isOpen=true', () => {
```

Comment:
I don't think we need this one, it should already be covered by the snapshots

Reply:
Or you could replace it with a "should open popover when clicking the reference element" (opposite of the test below, basically)